query-node/mappings/src/workingGroup.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+export async function storageWorkingGroup_TerminatedWorker(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId} = new StorageWorkingGroup.TerminatedWorkerEvent(event).data
+
+  // call generic processing
+  await workingGroup_TerminatedWorker(db, workerId)
+}
+
+export async function storageWorkingGroup_WorkerExited(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId} = new StorageWorkingGroup.WorkerExitedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerExited(db, workerId)
+}
+
+/////////////////// Gateway working group //////////////////////////////////////
+
+export async function gatewayWorkingGroup_OpeningFilled(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {applicationIdToWorkerIdMap} = new GatewayWorkingGroup.OpeningFilledEvent(event).data
+
+  // call generic processing
+  await workingGroup_OpeningFilled(db, WorkerType.GATEWAY, applicationIdToWorkerIdMap)
+}
+
+export async function gatewayWorkingGroup_WorkerStorageUpdated(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId, bytes: newMetadata} = new GatewayWorkingGroup.WorkerStorageUpdatedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerStorageUpdated(db, workerId, newMetadata)
+}
+
+export async function gatewayWorkingGroup_TerminatedWorker(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId} = new StorageWorkingGroup.TerminatedWorkerEvent(event).data
+
+  // call generic processing
+  await workingGroup_TerminatedWorker(db, workerId)
+}
+
+export async function gatewayWorkingGroup_WorkerExited(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {workerId} = new StorageWorkingGroup.WorkerExitedEvent(event).data
+
+  // call generic processing
+  await workingGroup_WorkerExited(db, workerId)
+}
+
+/////////////////// Generic working group processing ///////////////////////////
+
+export async function workingGroup_OpeningFilled(
+  db: DatabaseManager,
+  workerType: WorkerType,
+  applicationIdToWorkerIdMap: ApplicationIdToWorkerIdMap
+): Promise<void> {
+  const workerIds = [...applicationIdToWorkerIdMap.values()]
+
+  for (const workerId of workerIds) {
+    // create new worker
+    const newWorker = new Worker({
+      workerId: workerId.toString(),
+      type: workerType,
+      isActive: true,
+    })
+
+    await db.save<Worker>(newWorker)
+  }
+
+  // emit log event
+  logger.info("Worker has been created", {ids: workerIds.map(item => item.toString())})
+}
+
+export async function workingGroup_WorkerStorageUpdated(db: DatabaseManager, workerId: WorkerId, newMetadata: Bytes): Promise<void> {
+  // load worker
+  const worker = await db.get(Worker, { where: { id: workerId.toString() } as FindConditions<Worker> })
```

Comment:
I don't think this is correct. you are using worker id in place of the unique id

Reply:
Fixed